### PR TITLE
Excel: A sound is played when typing into protected cells rather than speaking typed characters

### LINKED ISSUE #6570: Speak typed characters and words in Excel in protected view (Excel 2016 at least)
When a spreadsheet is downloaded from the internet, Excel opens it in "Protected view" so that macros and other potentially malicious content cannot be run.  The workbook is also read only.

If you move to any cell in the workbook and type text, Excel ignores it - the original content of the cell is retained and visually, it never appears to go into edit mode (typed characters never appear anywhere).

With "speak typed characters" on, NVDA reads any characters typed twice.  With "speak typed words" on, NVDA, when spacebar is pressed, NVDA reads the word that was typed as individual letters.  For instance, type:

The

With speak typed characters on, NVDA reads:

T t h h e e

as you type.  When you press space, if speak typed characters is on, NVDA reads:

tthhee space space

With speak typed characters off, text is not read as you type it (but the "word" is still read if speak typed words is on).

The expected behaviour should be to alert the user in some manner that the text is not being typed (as a sighted user will see that the text is not being typed).

### PR DESCRIPTION
**Pull request by michaelDCurran**
Fixes #6570

### CODE REVIEW

**Comment by michaelDCurran** [comment]
@feerrenrut: review please.

File: source/NVDAObjects/window/excel.py [hunk 1]
```diff
@@ -1155,6 +1156,15 @@ def _get_states(self):
 			states.add(controlTypes.STATE_UNLOCKED)
 		return states
 
+	def event_typedCharacter(self,ch):
+		# # 6570: You cannot type into protected cells.
+		# A part from speaking characters being miss-leading, Office 2016 protected view doubles characters as well.
```
Comment (feerrenrut) [inline source/NVDAObjects/window/excel.py]: minor: "Apart" rather than "A part"

File: source/NVDAObjects/window/excel.py [hunk 2]
```diff
@@ -1155,6 +1156,15 @@ def _get_states(self):
 			states.add(controlTypes.STATE_UNLOCKED)
 		return states
 
+	def event_typedCharacter(self,ch):
+		# # 6570: You cannot type into protected cells.
```
Comment (feerrenrut) [inline source/NVDAObjects/window/excel.py]: minor: there is a space between the hash symbol and the issue number.

File: source/NVDAObjects/window/excel.py [hunk 3]
```diff
@@ -1155,6 +1156,15 @@ def _get_states(self):
 			states.add(controlTypes.STATE_UNLOCKED)
 		return states
 
+	def event_typedCharacter(self,ch):
+		# # 6570: You cannot type into protected cells.
+		# A part from speaking characters being miss-leading, Office 2016 protected view doubles characters as well.
+		# Therefore for any non-control char on protected cells, play the default sound rather than speaking the character
+		if ord(ch)>26 and controlTypes.STATE_UNLOCKED not in self.states and controlTypes.STATE_PROTECTED in self.parent.states:
```
Comment (feerrenrut) [inline source/NVDAObjects/window/excel.py]: It would be helpful if 26 was a named value. Is it the first printable character code?

**Comment by michaelDCurran** [comment]
How about this @feerrenrut? I addressed the comment issues, but I changed the numeric value into a space. I can add a constant if you think it is really necessary...

**Review by feerrenrut — APPROVED** [review]
Looks good to me.

**Pull request merged**